assertions/src/main/java/io/camunda/zeebe/process/test/inspections/model/InspectedProcessInstance.java

```diff
@@ -1,5 +1,18 @@
 package io.camunda.zeebe.process.test.inspections.model;
 
+/**
+ * Helper object to gain access to {@link
+ * io.camunda.zeebe.process.test.assertions.ProcessInstanceAssert}. All this object does is wrap a
+ * process instance key. This object is required to use {@link
+ * io.camunda.zeebe.process.test.assertions.BpmnAssert#assertThat(InspectedProcessInstance)}.
+ *
+ * <p>Without this object we would need a method {@code assertThat(long)}. This would conflict with
+ * the similar method in AssertJ, which is often imported in a static way.
+ *
+ * <p>Besides that it would also conflict with other assertions, such as {@link
+ * io.camunda.zeebe.process.test.assertions.IncidentAssert}, which could also be accessed using a
+ * long.
```

Comment:
🔧 This documentation should probably belong in a commit message. It describes why we took this approach over another. I'd rather see this documentation help a user how to use it.